packages/backend-core/src/users.ts

```diff
@@ -10,24 +10,52 @@ import { BulkDocsResponse, User } from "@budibase/types"
 import { getGlobalDB } from "./context"
 import * as context from "./context"
 
-export const bulkGetGlobalUsersById = async (userIds: string[]) => {
+type GetOpts = { cleanup?: boolean }
+
+function cleanupUsers(users: User | User[]) {
+  if (Array.isArray(users)) {
+    return users.map(user => {
+      if (user) {
+        delete user.password
+        return user
+      }
+    })
+  } else if (users) {
+    delete users.password
+    return users
+  }
+  return users
+}
+
+export const bulkGetGlobalUsersById = async (
+  userIds: string[],
+  opts?: GetOpts
+) => {
   const db = getGlobalDB()
-  return (
+  let users = (
     await db.allDocs({
       keys: userIds,
       include_docs: true,
     })
   ).rows.map(row => row.doc) as User[]
+  if (opts?.cleanup) {
+    users = cleanupUsers(users) as User[]
```

Comment:
The way we handle user passwords is a bit scary atm, it's feels too easy to read a user doc and accidentally delete the password by writing it again to the db without going through the proper sdk. 
We also do a lot of this in various places:
```ts
delete user.password
```

We'd be nearly better moving passwords out of the main user doc altogether, into the platform doc where we only write once. Could then have dedicated functions for password read / write, more protection around misuse by decoupling password updates from user saves.

Reply:
I agree, there were some issues that Gerard found around this - I would like to do something about this in the future, as this issue has come up a few times. I just created a slightly nicer layer to make it easier to remove it when returning users as an enrichment to handle this for the time being.